framework/Source/PictureInput.swift

```diff
@@ -56,32 +56,34 @@ public class PictureInput: ImageSource {
             }
         } else {
             let textureLoader = MTKTextureLoader(device: sharedMetalRenderingDevice.device)
-            if synchronously {
-                do {
-                    let mtlTexture = try textureLoader.newTexture(cgImage:internalImage!, options: newTextureOptions)
-                    internalImage = nil
-                    let texture = makeTexture(with: mtlTexture)
-                    self.internalTexture = texture
+            let semaphore: DispatchSemaphore? = synchronously ? DispatchSemaphore(value: 0) : nil
+            textureLoader.newTexture(cgImage: internalImage!, options: newTextureOptions, completionHandler: { [weak self] (possibleTexture, error) in
+                guard let self else {
+                    semaphore?.signal()
+                    return
+                }
+                if let error {
+                    semaphore?.signal()
+                    assertionFailure("Error in loading texture: \(error)")
+                    return
+
+                }
+                guard let mtlTexture = possibleTexture else {
+                    semaphore?.signal()
+                    assertionFailure("Nil texture received")
+                    return
+                }
+                self.internalImage = nil
+
+                let texture = makeTexture(with: mtlTexture)
+                self.internalTexture = texture
+                DispatchQueue.global().async{
```

Comment:
suggestion: we could improve by adding qos priority

Reply:
userInteractive?

Reply:
Hi @yyjim, I'm going to merge this PR first; please let me know if the userInteractive is not what you expected, and I'll update it in another PR